### PR TITLE
Eagerly evaluate `ifBreak` when processing template literals (fixes #795)

### LINKED ISSUE #795: Extra trailing commas in template literal.
Run prettier on: https://github.com/cpojer/jest/blob/master/packages/jest-cli/src/TestNamePatternPrompt.js#L91

```js
         pipe.write(
            // eslint-disable-next-line max-len
            `\n  ${chalk.dim(`\u203A and ${more} more ${pluralizeTest(more)}`)}`,
          );
```

turns it into this:

```js
          pipe.write(
            // eslint-disable-next-line max-len
            `\n  ${chalk.dim(`\u203A and ${more} more ${pluralizeTest(more,)}`,)}`,
          );
```

### PR DESCRIPTION
**Pull request by jlongster**
This should fix the problem with commas, but I wonder if we're going to run into anything else.

### CODE REVIEW

File: tests/strings/strings.js [hunk 1]
```diff
@@ -33,3 +33,7 @@ foo(`a long string ${ 1 + 2 + 3 + 2 + 3 + 2 + 3 + 2 + 3 + 2 + 3 + 2 + ( function
 
   return x;
  })() + 3 + 2 + 3 + 2 + 3 } with expr`);
+
+pipe.write(
```
Comment (vjeux) [inline tests/strings/strings.js]: btw, strings/strings.js was supposed to be around escaping stuff, not those template literal edge cases. Could you move those in another file/folder?

**Comment by vjeux** [comment]
Please move the tests in a different folder and that looks like it would solve it to me.

**Comment by jlongster** [comment]
Moved into a different file. I think `strings` is a nice directory :)
